packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -551,6 +551,11 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onColumnOrderChange?: GridEventListener<'columnOrderChange'>;
+  /**
+   * Fired when the density is updated.
```

Comment:
```suggestion
   * Callback fired when the density changes.
```

If we want to be consistent by props like `onFilterModelChange` or `onColumnVisibilityModelChange`

Maybe some props are using the wording "Fired when", it would be nice to unify at some point :ok_hand: